R/utils_data_handling.R

```diff
@@ -158,50 +158,113 @@ range_long_to_quantile <- function(data,
+#' be characterised by one or multiple prediction intervals, e.g. the lower
+#' and upper bounds of the 50% and 90% prediction intervals (corresponding to
+#' the 0.25 and 0.75 as well as the 0.05 and 0.095 quantiles).
+#' @param ... method arguments
+quantile_to_interval <- function(...) {
+  UseMethod("quantile_to_interval")
+}
+
+
+#' @param dt a data.table with columns `quantile` and `predicted`
+#' @param format the format of the output. Either "long" or "wide". If "long"
+#' (the default), there will be a column `boundary` (with values either "upper"
+#' or "lower" and a column `range` that contains the range of the interval.
+#' If "wide", there will be a column `range` and two columns
+#' `lower` and `upper` that contain the lower and upper bounds of the
+#' prediction interval, respectively.
 #' @param keep_quantile_col keep the quantile column in the final
-#' output after transformation (default is FALSE)
-#' @return a data.frame in a long interval range format
+#' output after transformation (default is FALSE). This only works if
+#' `format = "long"`. If `format = "wide"`, the quantile column will always be
+#' dropped.
+#' @return
+#' *quantile_to_interval.data.frame*:
+#' a data.frame in an interval format (either "long" or "wide"), with or
+#' without a quantile column. Rows will not be reordered.
 #' @importFrom data.table copy
-#' @keywords internal
-
-quantile_to_range_long <- function(data,
-                                   keep_quantile_col = TRUE) {
-  data <- data.table::as.data.table(data)
+#' @export
+#' @rdname quantile_to_interval
+quantile_to_interval.data.frame <- function(dt,
+                                            format = "long",
+                                            keep_quantile_col = FALSE,
+                                            ...) {
+  if (!is.data.table(dt)) {
+    dt <- data.table::as.data.table(dt)
+  } else {
+    # use copy to avoid
+    dt <- copy(dt)
+  }
 
-  data[, boundary := ifelse(quantile <= 0.5, "lower", "upper")]
-  data[, range := ifelse(
+  dt[, boundary := ifelse(quantile <= 0.5, "lower", "upper")]
+  dt[, range := ifelse(
     boundary == "lower",
     round((1 - 2 * quantile) * 100, 10),
     round((2 * quantile - 1) * 100, 10)
   )]
 
   # add median quantile
-  median <- data[quantile == 0.5, ]
+  median <- dt[quantile == 0.5, ]
   median[, boundary := "upper"]
 
-  data <- data.table::rbindlist(list(data, median))
-
+  dt <- data.table::rbindlist(list(dt, median))
   if (!keep_quantile_col) {
-    data[, "quantile" := NULL]
+    dt[, quantile := NULL]
   }
 
-  # if only point forecasts are scored, we only have NA values for range and
-  # boundary. In that instance we need to set the type of the columns
-  # explicitly to avoid future collisions.
-  data[, `:=`(
-    boundary = as.character(boundary),
-    range = as.numeric(range)
-  )]
+  if (format == "wide") {
+    delete_columns(dt, "quantile")
```

Comment:
is this internal function really needed?

Reply:
Not anymore I think. `data.table` used to error if you deleted a column that didn't exist. Now it throws a warning. 
We could just do `suppresWarnings(dt[, "quantile" := NULL]` instead

Reply:
Made a discussion issue, #446